### PR TITLE
[WFAPI-1303] Added ability to publish to a specific topic

### PR DESCRIPTION
**Pull request by emirkin**
Previously, there was no way to specify the destination topic (if anything other than "self" was desired). No longer.

### CODE REVIEW

File: lib/pub_sub/publisher.rb [hunk 1]
```diff
@@ -33,15 +33,12 @@ def sns
...
       end
 
-      def topic_arn
+      def topic_arn(topic_name)
         Breaker.run do
           sns.create_topic(name: topic_name).topic_arn
         end
       end
 
-      def topic_name
```
Comment (ashoda) [inline lib/pub_sub/publisher.rb]: If the above are `custom_topic`s don't we want to have a default topic?

**Comment by ashoda** [comment]
Out of curiosity what is the use-case for this at the moment of moving forward?

File: lib/pub_sub/publisher.rb [hunk 2]
```diff
@@ -1,28 +1,28 @@
 module PubSub
   class Publisher
     class << self
-      def publish(message, async: false)
+      def publish(message, async: false, custom_topic: PubSub.service_identifier)
```
Comment (ashoda) [inline lib/pub_sub/publisher.rb]: I see that it starts with a default. here :point_up_2: . Do we want to call this `custom_topic` seems like topic should suffice.

File: lib/pub_sub/publisher.rb (hunk 1 above)
Comment (emirkin) [inline lib/pub_sub/publisher.rb]: We do have a default topic up above - at `publish` level

**Comment by emirkin** [comment]
It gives publishers the ability to post to topics other than their own (of which there can be only one)

File: lib/pub_sub/publisher.rb (hunk 2 above)
Comment (ashoda) [inline lib/pub_sub/publisher.rb]: @emirkin

**Comment by ashoda** [comment]
@emirkin That part is straight forward. I was asking why or when do we need to do that. Is there a use case at the moment where one service needs to publish on other topics?

**Pull request merged**